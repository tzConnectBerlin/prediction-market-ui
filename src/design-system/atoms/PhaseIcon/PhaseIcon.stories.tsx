import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { PreTradingIcon, TradingIcon, ResolvedIcon } from './PhaseIcon';

export default {
  title: 'Atom/PhaseIcon',
  component: PreTradingIcon,
} as Meta;

const Template: Story = () => <PreTradingIcon />;
export const PreTrading = Template.bind({});

const TradingTemplate: Story = () => <TradingIcon />;
export const Trading = TradingTemplate.bind({});

const ResolvedTemplate: Story = () => <ResolvedIcon />;
export const Resolved = ResolvedTemplate.bind({});
