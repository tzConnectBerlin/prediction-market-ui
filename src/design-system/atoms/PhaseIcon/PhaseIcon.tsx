import * as React from 'react';
import { useTheme } from '@material-ui/core';
import FiberManualRecord from '@material-ui/icons/FiberManualRecord';
import styled from '@emotion/styled';

const StyledPhaseIcon = styled(FiberManualRecord)<{ iconColor?: string }>`
  width: 0.9375rem;
  height: 0.9375rem;
  margin: 0.375rem 0.9375rem 0.375rem 0;
  border-radius: 2.75rem;
  padding: 0;
  min-width: 0;
  border: solid 1px rgba(29, 34, 39, 0.04);
  & circle {
    r: 12;
  }
  color: ${({ iconColor }) => iconColor};
`;

export interface PhaseIconProps {
  variant?: 'primary' | 'secondary' | 'tertiary';
}

export const PhaseIcon: React.FC<PhaseIconProps> = ({ variant = 'primary' }) => {
  const theme = useTheme();
  const color =
    variant === 'primary'
      ? theme.palette.primary.light
      : variant === 'secondary'
      ? theme.palette.primary.main
      : theme.palette.grey[400];
  return <StyledPhaseIcon iconColor={color} />;
};
