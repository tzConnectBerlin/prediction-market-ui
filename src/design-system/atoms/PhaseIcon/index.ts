export { PhaseIcon } from './PhaseIcon';
export type { PhaseIconProps } from './PhaseIcon';
