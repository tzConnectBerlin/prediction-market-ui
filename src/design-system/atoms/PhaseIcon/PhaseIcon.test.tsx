import renderer from 'react-test-renderer';
import { PhaseIcon } from './PhaseIcon';

describe('Snapshot Testing TwitterShare Component', () => {
  it('renders correctly with blue color', () => {
    const PreTrading = renderer.create(<PhaseIcon variant="primary" />).toJSON();
    expect(PreTrading).toMatchSnapshot();
  });

  it('renders correctly with grey color', () => {
    const Trading = renderer.create(<PhaseIcon variant="secondary" />).toJSON();
    expect(Trading).toMatchSnapshot();
  });

  it('renders correctly without title', () => {
    const Resolved = renderer.create(<PhaseIcon variant="tertiary" />).toJSON();
    expect(Resolved).toMatchSnapshot();
  });
});
