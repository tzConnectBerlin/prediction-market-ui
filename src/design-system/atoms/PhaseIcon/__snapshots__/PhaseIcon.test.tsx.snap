// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing TwitterShare Component renders correctly with blue color 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 0.9375rem;
  height: 0.9375rem;
  margin: 0.375rem 0.9375rem 0.375rem 0;
  border-radius: 2.75rem;
  padding: 0;
  min-width: 0;
  border: solid 1px rgba(29, 34, 39, 0.04);
  color: #42a5f5;
}

.emotion-0 circle {
  r: 12;
}

<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
  data-testid="FiberManualRecordIcon"
  focusable="false"
  iconColor="#42a5f5"
  viewBox="0 0 24 24"
>
  <circle
    cx="12"
    cy="12"
    r="8"
  />
</svg>
`;

exports[`Snapshot Testing TwitterShare Component renders correctly with grey color 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 0.9375rem;
  height: 0.9375rem;
  margin: 0.375rem 0.9375rem 0.375rem 0;
  border-radius: 2.75rem;
  padding: 0;
  min-width: 0;
  border: solid 1px rgba(29, 34, 39, 0.04);
  color: #1976d2;
}

.emotion-0 circle {
  r: 12;
}

<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
  data-testid="FiberManualRecordIcon"
  focusable="false"
  iconColor="#1976d2"
  viewBox="0 0 24 24"
>
  <circle
    cx="12"
    cy="12"
    r="8"
  />
</svg>
`;

exports[`Snapshot Testing TwitterShare Component renders correctly without title 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 0.9375rem;
  height: 0.9375rem;
  margin: 0.375rem 0.9375rem 0.375rem 0;
  border-radius: 2.75rem;
  padding: 0;
  min-width: 0;
  border: solid 1px rgba(29, 34, 39, 0.04);
  color: #bdbdbd;
}

.emotion-0 circle {
  r: 12;
}

<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
  data-testid="FiberManualRecordIcon"
  focusable="false"
  iconColor="#bdbdbd"
  viewBox="0 0 24 24"
>
  <circle
    cx="12"
    cy="12"
    r="8"
  />
</svg>
`;
